New version: Deskflow.Deskflow version 1.20.1

nbolton opened the pull request:
Checklist for Pull Requests
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.9 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.9.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/235708)

nbolton committed:
New version: Deskflow.Deskflow version 1.20.1

stephengillie commented:
Automatic Validation ended with:
>  Installation failed with exit code -1978335226 2025-03-07 09:30:41.711 [CLI ] ShellExecute installer failed: 1603 Action start 9:30:41: Wix5SchedFirewallExceptionsUninstall_X64. SchedFirewallExceptions:  Component 'CM_CP_deskflow_client.exe' action state (1) doesn't match request (2) SchedFirewallExceptions:  Component 'CM_CP_deskflow_server.exe' action state (1) doesn't match request (2) SchedFirewallExceptions:  No firewall exceptions scheduled Action ended 9:30:41: Wix5SchedFirewallExceptionsUninstall_X64. Return value 1. Action start 9:30:41: Wix5SchedFirewallExceptionsInstall_X64. SchedFirewallExceptions:  Scheduling firewall exception (1€Deskflow Client€-2147483648€*€0€C:\Program Files\Deskflow\deskflow-client.exe€€-2147483648€€1€-2147483648€-2147483648€-2147483648€€€€€€€€€€€€€-2147483648€1€Deskflow Server€-2147483648€*€0€C:\Program Files\Deskflow\deskflow-server.exe€€-2147483648€€1€-2147483648€-2147483648€-2147483648€€€€€€€€€€€€€-2147483648) Action start 9:30:41: Wix5RollbackFirewallExceptionsInstall_X64. Action ended 9:30:41: Wix5RollbackFirewallExceptionsInstall_X64. Return value 1. Action start 9:30:41: Wix5ExecFirewallExceptionsInstall_X64. Action ended 9:30:41: Wix5ExecFirewallExceptionsInstall_X64. Return value 1. Action ended 9:30:41: Wix5SchedFirewallExceptionsInstall_X64. Return value 1. Action start 9:30:41: ShowVCRedistError. Action ended 9:30:41: ShowVCRedistError. Return value 3. Property(S): ErrorDialog = ErrorDlg Property(S): Wix5RollbackFirewallExceptionsInstall_X64 = 1Deskflow Client-2147483648*0C:\Program Files\Deskflow\deskflow-client.exe-21474836481-2147483648-2147483648-2147483648-21474836481Deskflow Server-2147483648*0C:\Program Files\Deskflow\deskflow-server.exe-21474836481-2147483648-2147483648-2147483648-2147483648 Property(S): Wix5ExecFirewallExceptionsInstall_X64 = 1Deskflow Client-2147483648*0C:\Program Files\Deskflow\deskflow-client.exe-21474836481-2147483648-2147483648-2147483648-21474836481Deskflow Server-2147483648*0C:\Program Files\Deskflow\deskflow-server.exe-21474836481-2147483648-2147483648-2147483648-2147483648 Property(S): OutOfDiskSpace = 0 Property(S): OutOfNoRbDiskSpace = 0 Property(S): PrimaryVolumeSpaceAvailable = 0 Property(S): PrimaryVolumeSpaceRequired = 0 Property(S): PrimaryVolumeSpaceRemaining = 0 MSI (s) (98:E4) [09:30:41:396]: Product: Deskflow -- Installation failed.

Automated error analysis suggests `-1978335226` (``) may translate to token `` which may indicate "Running ShellExecute failed"

Automated error analysis suggests `1603` (``) may translate to token `` which may indicate "Previous MSI installer is missing (Or it might mean Install successful but system restart may be required.) Should this be add as an `InstallerSuccessCode` or similar?"

(Automated response - build 991.)

nbolton commented:
@nbolton stephengillie We added a check in the MSI for VC++ redist. So that'll be why this is failing.

Changed files: https://github.com/deskflow/deskflow/pull/8327/files

Do we need to add something to the winget manifest?

stephengillie commented:
Manual Validation ended with:
> 2:16:24 PM: Product: Deskflow -- Latest Microsoft Visual C++ Redistributable is required. Please install it before proceeding.

(Automated response - build 994.)

stephengillie commented on manifests/d/Deskflow/Deskflow/1.20.1/Deskflow.Deskflow.installer.yaml:
```diff
@@ -0,0 +1,14 @@
+InstallerLocale: en-US
+InstallerType: wix
+ProductCode: '{3BC8942A-82C2-491D-AC2F-3B4B4B0B775C}'
+Installers:
```
```suggestion
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.VCRedist.2015+.x64
Installers:
```

stephengillie replied:
It appears to:

```PowerShell
PS C:\ManVal> Find-WinGetPackage Microsoft.VCRedist.2015+.x64

Name                                                 Id                           Version       Source
----                                                 --                           -------       ------
Microsoft Visual C++ 2015-2022 Redistributable (x64) Microsoft.VCRedist.2015+.x64 14.42.34438.0 winget
```

nbolton committed:
fix: Add `Microsoft.VCRedist.2015+.x64` dependency

Co-authored-by: Stephen Gillie <114952565+stephengillie@users.noreply.github.com>
